lib/resources/properties.js

```diff
@@ -0,0 +1,123 @@
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is furnished to do
+ * so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+
+/**
+ * Extends the client object to include resource routes
+ * @param {Client} Client
+ * @private
+ */
+function registerMethods(Client) {
+
+    /**
+     * Session Methods
+     * @namespace Client.properties
+     */
+    Client.properties = {
+
+        /**
+         * Creates a new property.
+         * @param {object} params
+         * @param {requestCallback} callback
+         * @memberof Client.properties#
+         */
+        create: function(params, callback) {
+            return Client._makeRequest({
+                method: 'POST',
+                path: '/properties',
+                payload: params
+            }, callback);
+        },
+
+        /**
+         * Retrieves a property.
+         * @param {string} propertyId
+         * @param {requestCallback} callback
+         * @memberof Client.properties#
+         */
+        retrieve: function(propertyId, callback) {
+            return Client._makeRequest({
+                method: 'GET',
+                path: '/properties/{propertyId}',
+                pathParams: {
+                    propertyId: propertyId
+                }
+            }, callback);
+        },
+
+        /**
+         * Lists properties.
+         * @param [params] Query filter criteria
+         * @param {requestCallback} callback
+         * @memberof Client.properties#
+         */
+        list: function(params, callback) {
+            if (typeof params === "function") {
+                callback = params;
+                params = undefined;
+            }
+
+            return Client._makeRequest({
+                method: 'GET',
+                path: '/properties',
+                query: params
+            }, callback);
+        },
+
+        /**
+         * Updates a property.
+         * @param {string} propertyId
+         * @param {object|null} params
+         * @param {requestCallback} callback
+         * @memberof Client.properties#
+         */
+
+        update: function(propertyId, params, callback) {
+            return Client._makeRequest({
```

Comment:
I think there should be a check here for optional params / callback - if the user wants to do `api.properties(id).data(asdf).etc`